src/libs/actions/IOU.ts

```diff
@@ -6515,8 +6554,8 @@ function getReportFromHoldRequestsOnyxData(
         chatReport.reportID,
         chatReport.policyID ?? iouReport?.policyID ?? '',
         recipient.accountID ?? 1,
-        holdTransactions.reduce((acc, transaction) => acc + TransactionUtils.getAmount(transaction), 0),
-        getCurrency(firstHoldTransaction),
+        ((iouReport?.total ?? 0) - (iouReport?.unheldTotal ?? 0)) * (ReportUtils.isIOUReport(iouReport) ? 1 : -1),
+        iouReport?.currency ?? '',
         false,
         newParentReportActionID,
```

Comment:
Probably as a follow up: I noticed we pass `reimbursable=false` when creating a new report (to move the held expenses). This is not exactly correct because the held expenses can be mixed (reimbursable and nonReimbursable) which means we also need to fix the calculated (nonReimbursableTotal) value in `buildOptimisticExpenseReport`.

Reply:
Nice catch, I think we can update it in this PR. There is only 2 usages of `buildOptimisticExpenseReport` that passes the `reimbursable` param. I think we can change it to nonReimbursableTotal param instead.

Reply:
Done